content-src/lib/selectLayoutRender.js

```diff
@@ -2,18 +2,29 @@ export const selectLayoutRender = (state, rickRollCache) => {
   const {layout, feeds, spocs} = state;
   let spocIndex = 0;
 
+  // rickRollCache stores random probability values for each spoc position. This cache is empty
+  // on page refresh and gets filled with random values on first render inside maybeInjectSpocs.
+  const isFirstRun = !rickRollCache.length;
+  // rickRollCache stores values in the order components are rendered
+  // to be used in subsequent renders for consuming by reversing array.
+  rickRollCache.reverse();
+
   function maybeInjectSpocs(data, spocsConfig) {
     if (data &&
         spocsConfig && spocsConfig.positions && spocsConfig.positions.length &&
         spocs.data.spocs && spocs.data.spocs.length) {
       const recommendations = [...data.recommendations];
       for (let position of spocsConfig.positions) {
         // Cache random number for a position
-        if (!rickRollCache[position.index]) {
-          rickRollCache[position.index] = Math.random();
+        let rickRoll;
+        if (isFirstRun) {
+          rickRoll = Math.random();
+          rickRollCache.push(rickRoll);
+        } else {
+          rickRoll = rickRollCache.pop();
```

Comment:
I think if you used `array.shift();` here instead of pop, you wouldn't need the `rickRollCache.reverse();` above. I *think*

Reply:
After grabbing a reference to the roll here, I think you should also push it to the buffer array. Then when you finish your final maybeInjectSpocs you just reset the cache array (which should now be empty again) with the buffer that's in this local scope (which should now be full)